release/benchmarks/distributed/dashboard_test.py

```diff
@@ -0,0 +1,144 @@
+    "/logical/actors",
+    "/nodes?view=summary",
+    "/",
+    "/api/cluster_status",
+    "/events",
+    "/api/jobs/",
+    "/log_index",
+    "/api/prometheus_health",
+]
+
+
+@ray.remote(num_cpus=0)
+class DashboardTester:
+    def __init__(self, addr: ray._private.worker.RayContext, interval_s: int = 1):
+        self.dashboard_url = format_web_url(addr.dashboard_url)
+        # Ping interval for all endpoints.
+        self.interval_s = interval_s
+        # endpoint -> a list of latencies
+        self.result = defaultdict(list)
+
+    async def run(self):
+        await asyncio.gather(*[self.ping(endpoint) for endpoint in endpoints])
+
+    async def ping(self, endpoint):
+        """Synchronously call an endpoint."""
+        while True:
+            start = time.monotonic()
+            resp = requests.get(self.dashboard_url + endpoint)
+            elapsed = time.monotonic() - start
+
+            if resp.status_code == 200:
+                self.result[endpoint].append(time.monotonic() - start)
+            else:
+                try:
+                    resp.raise_for_status()
+                except Exception as e:
+                    logger.exception(e)
+            await asyncio.sleep(max(0, self.interval_s, elapsed))
+
+    def get_result(self):
+        return self.result
+
+
+class DashboardTestAtScale:
+    """This is piggybacked into existing scalability tests."""
+
+    def __init__(self, addr: ray._private.worker.RayContext):
+        self.addr = addr
+
+        # Schedule the actor on the current node (which is a head node).
+        current_node_ip = ray._private.worker.global_worker.node_ip_address
+        nodes = ray.experimental.state.api.list_nodes()
+        node = list(filter(lambda n: n["node_ip"] == current_node_ip, nodes))[0]
+        # Schedule on a head node.
+        self.tester = DashboardTester.options(
+            scheduling_strategy=NodeAffinitySchedulingStrategy(
+                node_id=node["node_id"], soft=False
+            )
+        ).remote(addr)
+
+        self.tester.run.remote()
+
+    def get_result(self):
+        """Get the result from the test.
+
+        Returns:
+            A tuple of success, and the result (Result object).
+        """
+        try:
+            result = ray.get(self.tester.get_result.remote(), timeout=60)
+        except ray.exceptions.GetTimeoutError:
+            return Result(success=False)
+
+        def calc_p(percent):
+            return {
+                # sort -> get PX -> convert second to ms -> round up.
+                endpoint: round(
+                    sorted(latencies)[int(len(latencies) / 100 * percent)] * 1000, 3
+                )  # noqa
+                for endpoint, latencies in result.items()
+            }
+
+        # Get the memory usage.
+        dashboard_export_addr = "{}:{}".format(
+            self.addr["raylet_ip_address"], DASHBOARD_METRIC_PORT
+        )
+        metrics = fetch_prometheus_metrics([dashboard_export_addr])
```

Comment:
Q: Ideally, we should query prometheus directly and get max_over_time. Should we go this way?